File: Algorithms/69.sqrtx/sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        if (x <= 0) return 0;

    int lb = 0, ub = x;
    while (lb + 1 < ub) {
        long long mid = lb + (ub - lb) / 2; 
        if (mid * mid == x) return mid; 
        if (mid * mid < x) lb = mid;
        else ub = mid;
    }
    return lb;
    }
};

//有Bug

File: Algorithms/69.sqrtx/sqrtx.py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x < 0:
            return -1
        elif x == 0:
            return 0

        start, end = 1, x
        while start + 1 < end:
            mid = start + (end - start) / 2
            if mid**2 == x:
                return mid
            elif mid**2 > x:
                end = mid
            else:
                start = mid

        return start
